doc/policy/mempool-replacements.md

```diff
@@ -0,0 +1,65 @@
+# Mempool Replacements
+
+## Current Replace-by-Fee Policy
+
+A transaction conflicts with an in-mempool transaction ("directly conflicting transaction") if they
+spend one or more of the same inputs. A transaction may conflict with multiple in-mempool
+transactions.
+
+A transaction ("replacement transaction") may replace its directly conflicting transactions and
+their in-mempool descendants (together, "original transactions") if, in addition to passing all
+other consensus and policy rules, each of the following conditions are met:
+
+1. The directly conflicting transactions must all signal replaceability explicitly. A transaction is
+   signaling replaceability if any of its inputs have an nSequence number less than (0xffffffff - 1).
+
+   *Rationale*: See [BIP125
+   explanation](https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#motivation).
+
+2. The replacement transaction may only include an unconfirmed input if that input was included in
+   one of the directly conflicting transactions. An unconfirmed input spends an output from a
+   currently-unconfirmed transaction.
+
+   *Rationale*: When RBF was originally implemented, the mempool did not keep track of
+   ancestor scores yet. This rule was suggested as a temporary restriction.
+
+3. The replacement transaction must pay an absolute fee of at least the sum paid by the original
+   transactions.
+
+   *Rationale*: Only requiring the replacement transaction to have a higher feerate could allow an
+   attacker to bypass node minimum relay feerate requirements and cause the network to repeatedly
+   relay slightly smaller replacement transactions without adding any more fees.
+
+4. The additional fees (difference between absolute fee paid by the replacement transaction and the
+   sum paid by the original transactions) must pay for the replacement transaction's bandwidth at or
+   above the rate set by the node's incremental relay feerate. For example, if the incremental relay
+   feerate is 1 satoshi/vB and the replacement transaction is 500 virtual bytes total, then the
+   replacement must pay a fee at least 500 satoshis higher than the sum of the original transactions.
+
+   *Rationale*: Try to attempt DoS attacks where an attacker causes the network to repeatedly relay
+   transactions each paying a tiny additional amount in fees, e.g. just 1 satoshi.
+
+5. The number of original transactions must not exceed 100. More precisely, the sum of all
+   directly conflicting transactions' descendant counts (number of transactions inclusive of itself
+   and its descendants) must not exceed 100.
+
+   *Rationale*: Try to prevent DoS attacks where an attacker is able to easily occupy and flush out
+   significant portions of the node's mempool using replacements with multiple directly conflicting
+   transactions, each with large descendant sets.
+
+This set of rules is similar but distinct from BIP125.
```

Comment:
I'm not sure I understand what you mean here by "similar but distinct" ? Maybe say we're partial implementation of BIP125 as we have at least one known divergence (though maybe fixed by #22698)

Reply:
I think there's a lot of differences and it may change over time, so it's better to just say they're distinct than try to convey a measurement of how different they are. I also don't like the framing of "partial implementation", since the goal was never to implement BIP125. If anything, BIP125 is a partial description of our code.

More differences:
- We use the node's incrementalRelayFee, not minrelayfeerate now, so if the user sets something different for those two values, we're diverging from BIP125
- We use the sum of directly conflicting transactions' descendant counts, not the exact number of original transactions.

Reply:
> since the goal was never to implement BIP125

Yes BIP125 was written from Core's implementation. Thanks for the explanation, I agree :)